core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -340,6 +353,44 @@ public void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploym
         }
     }
 
+    /**
+     * Execute given test scenario.
+     *
+     * @param testScenario the test scenario
+     * @param deploymentCreationResult the result of the previous build step
+     * @param testPlan the test plan
+     */
+    private void executeTestScenario(TestScenario testScenario, DeploymentCreationResult deploymentCreationResult,
+                                     TestPlan testPlan) {
+        try {
+            scenarioExecutor.execute(testScenario, deploymentCreationResult, testPlan);
+            Optional<ResultParser> parser = ResultParserFactory.getParser(testPlan, testScenario);
+            if (parser.isPresent()) {
+                try {
+                    ResultParser resultParser = parser.get();
+                    resultParser.parseResults();
+                    resultParser.archiveResults();
+                } catch (ResultParserException e) {
+                    logger.error(String.format("Error parsing the results for the" +
+                            " scenario %s", testScenario.getName()));
+                }
+            } else {
+                testScenario.setStatus(Status.ERROR);
+                logger.error(String.format("Error parsing the results for the " +
+                        "scenario %s", testScenario.getName()));
+            }
+        } catch (ScenarioExecutorException e) {
+            // we need to continue the rest of the tests irrespective of the exception thrown here.
+            logger.error(StringUtil.concatStrings("Error occurred while executing the SolutionPattern '",
+                    testScenario.getName(), "' in TestPlan\nCaused by "), e);
+        }
+        try {
+            persistTestScenario(testScenario);
+        } catch (TestPlanExecutorException e) {
+            logger.error(StringUtil.concatStrings(
+                    "Error occurred while persisting test scenario ", testScenario.getName()), e);
```

Comment:
Do we need to throw the exception to the object whom calling this?

Reply:
Catching exceptions and logging them is not considered a bad practice!